spec/message.abnf

```diff
@@ -14,9 +14,9 @@ local-declaration = local s variable [s] "=" [s] expression
 
 quoted-pattern    = "{{" pattern "}}"
 
-matcher           = match-statement 1*([s] variant)
-match-statement   = match 1*([s] selector)
-selector          = expression
+matcher           = match-statement s variant *([s] variant)
+match-statement   = match 1*(s selector)
+selector          = variable
```

Comment:
Perhaps less complicated as:

```suggestion
matcher           = match-statement s 1*(variant [s])
match-statement   = match 1*(s selector)
selector          = variable
```

Reply:
While that works, it would mean that it's not clear if trailing spaces at the end of a complex message would belong to the `matcher` or the `complex-message`.

Reply:
The spaces are superfluous to both matcher and complex-message, so perhaps okay? But what you have works.